HPXMLtoOpenStudio/resources/location.rb

```diff
@@ -60,17 +62,25 @@ def self.apply_year(model, epw_file)
     end
   end
 
-  def self.apply_dst(model, dst_start_date, dst_end_date)
-    return if dst_start_date.nil? || dst_end_date.nil?
-
-    dst_start_date_month = OpenStudio::monthOfYear(dst_start_date.split[0])
-    dst_start_date_day = dst_start_date.split[1].to_i
-    dst_end_date_month = OpenStudio::monthOfYear(dst_end_date.split[0])
-    dst_end_date_day = dst_end_date.split[1].to_i
+  def self.apply_dst(model, epw_file, hpxml)
+    month_names = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']
+
+    if (not @dst_begin_month.nil?) && (not @dst_begin_day_of_month.nil?) && (not @dst_end_month.nil?) && (not @dst_end_day_of_month.nil?)
+      dst_start_date = "#{month_names[@dst_begin_month]} #{@dst_begin_day_of_month}"
+      dst_end_date = "#{month_names[@dst_end_month]} #{@dst_end_day_of_month}"
+    elsif epw_file.daylightSavingStartDate.is_initialized && epw_file.daylightSavingEndDate.is_initialized
+      dst_start_date = epw_file.daylightSavingStartDate.get
+      dst_start_date = "#{month_names[dst_start_date.monthOfYear.value - 1]} #{dst_start_date.dayOfMonth}"
+      dst_end_date = epw_file.daylightSavingEndDate.get
+      dst_end_date = "#{month_names[dst_end_date.monthOfYear.value - 1]} #{dst_end_date.dayOfMonth}"
+    else
+      dst_start_date = 'Mar 11'
+      dst_end_date = 'Nov 4'
+    end
```

Comment:
Move this code to hpxml_defaults.rb, so that the chosen start/end date can show up in the in.xml file for inspection. And then add checks in test_defaults.rb.

Reply:
Can you explain this further?

Reply:
When the DST values are not provided, they should be defaulted [https://github.com/NREL/OpenStudio-HPXML/blob/master/HPXMLtoOpenStudio/resources/hpxml_defaults.rb#L35-L41](here) similar to simulation control start/end dates.

Reply:
So pass `epw_file` into the `HPXMLDefaults` class?

Reply:
Yes.